### ISSUE TITLE
cuda compatibility with torch in docker

### ISSUE THREAD

**Issue by lleye** [ultralytics/yolov5#4027]
pulled updated docker image but getting errors below. 

A100-SXM4-40GB with CUDA capability sm_80 is not compatible with the current PyTorch installation.
The current PyTorch install supports CUDA capabilities sm_37 sm_50 sm_60 sm_70.
If you want to use the A100-SXM4-40GB GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/

i did this in env before running docker: `pip install torch==1.9.0+cu111 torchvision==0.10.0+cu111 -f https://download.pytorch.org/whl/torch_stable.html`

`pip show torch` Version: 1.9.0

@glenn-jocher @nocolour

**Comment by glenn-jocher** [comment]
@lleye yes for some configurations you may need to visit https://pytorch.org/get-started/locally/

**Comment by iceisfun** [comment]
I run with 4x A100-SXM4-40GB using Dockerfile FROM nvcr.io/nvidia/pytorch:21.06-py3 and everything works fine

Also...
NVIDIA-SMI 460.73.01    Driver Version: 460.73.01    CUDA Version: 11.2
root@container:/usr/src/app# pip3 show torch
Name: torch
Version: 1.9.0a0+c3d40fd

And docker ....  --ipc=host --gpus all -v /sys:/sys  ....


@lleye have you done any testing on A100 vs non A100 platforms for speed?

I'm curious as I only have a single A100 machine and it performs pretty bad compared to RTX Titan or RTX 3090 based machines for training and inferance.

**Comment by glenn-jocher** [comment]
@iceisfun if your A100 speeds are similar to your 3090 speeds then I would say everything is clearly not working fine. 

I would recommend the official image with updates from https://pytorch.org/get-started/locally/ as necessary.

**Comment by lleye** [comment]
@iceisfun A100 is much faster, there is almost a 2x speedup compared v100s. convergence and accuracy is also better since you can use a larger batch size and do other stuff. yes, i do --gpus all etc

@glenn-jocher okay, will try to configure from pytorch local and see if it gets resolved

**Comment by Pro100rus32** [comment]
It's bug. Cuda 11 slower than cuda 10 in pytorch

**Comment by lleye** [comment]
`pip3 install torch==1.9.0+cu111  -f https://download.pytorch.org/whl/torch_stable.html` resolves the issue but then there is an nccl error as below for multi gpu training.

[W ProcessGroupNCCL.cpp:1569] Rank 0 using best-guess GPU 0 to perform barrier as devices used by this process are currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect.Specify device_ids in barrier() to force use of a particular device.

and 

[E ProcessGroupNCCL.cpp:325] Some NCCL operations have failed or timed out. Due to the asynchronous nature of CUDA kernels, subsequent GPU operations might run on corrupted/incomplete data. To avoid this inconsistency, we are taking the entire process down.
terminate called after throwing an instance of 'std::runtime_error'

**Comment by xiaohu2015** [comment]
@lleye I also got this warning, I also find A100 on torch1.9 is much slow than A100 on torch1.7

**Comment by guillaumefrd** [comment]
> `pip3 install torch==1.9.0+cu111 -f https://download.pytorch.org/whl/torch_stable.html` resolves the issue but then there is an nccl error as below for multi gpu training.
> 
> [W ProcessGroupNCCL.cpp:1569] Rank 0 using best-guess GPU 0 to perform barrier as devices used by this process are currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect.Specify device_ids in barrier() to force use of a particular device.
> 
> and
> 
> [E ProcessGroupNCCL.cpp:325] Some NCCL operations have failed or timed out. Due to the asynchronous nature of CUDA kernels, subsequent GPU operations might run on corrupted/incomplete data. To avoid this inconsistency, we are taking the entire process down. terminate called after throwing an instance of 'std::runtime_error'

I have the same error using 4 V100-SXM2-16GB, torch 1.9.0+cu111 and CUDA 11.1. It happens at the end of the first epoch.

**Comment by Shuweis** [comment]
> > `pip3 install torch==1.9.0+cu111 -f https://download.pytorch.org/whl/torch_stable.html` resolves the issue but then there is an nccl error as below for multi gpu training.
> > [W ProcessGroupNCCL.cpp:1569] Rank 0 using best-guess GPU 0 to perform barrier as devices used by this process are currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect.Specify device_ids in barrier() to force use of a particular device.
> > and
> > [E ProcessGroupNCCL.cpp:325] Some NCCL operations have failed or timed out. Due to the asynchronous nature of CUDA kernels, subsequent GPU operations might run on corrupted/incomplete data. To avoid this inconsistency, we are taking the entire process down. terminate called after throwing an instance of 'std::runtime_error'
> 
> I have the same error using 4 V100-SXM2-16GB, torch 1.9.0+cu111 and CUDA 11.1. It happens at the end of the first epoch.

Hello，I have the same error.Have you solved it？

**Comment by zzm-hl** [comment]
have you solved the problem ?

**Comment by glenn-jocher** [comment]
@zzm-hl the issue you are encountering may be related to the NCCL backend for multi-GPU training. The problem can stem from various sources such as mismatched NCCL versions or incorrect GPU mapping. We recommend checking your NCCL installation, ensuring consistent versions among all GPUs, and verifying the GPU mapping.

For further assistance with resolving this issue, we suggest referring to the PyTorch and NCCL documentation. If the issue persists, feel free to provide additional details about your setup so we can offer more specific help.

Thank you for your understanding and patience while we work through this together.
